File: LinkedList/LinkedList/revision/bst.cpp
#include <iostream>
#include <queue>
using namespace std;


struct Node{
    Node* left;
    int data;
    Node* right;

    Node(int value) : left(nullptr), data(value), right(nullptr) {}
};

class BST{
    private:
        Node* buildTree(Node* node, int value){
            if(node == nullptr)
                return new Node(value);
            
            Node* parent = new Node(value);

            if(node->data > value)
                node->left = parent;
            else:
                node->right = parent;

            return parent;
        }
    public:
        Node* root;

        BST() : root(nullptr) {}

        void insert(){
            int input;
            cin >> input;

            while(input != -1){
                root = buildTree(root, input);
                cin >> input;
            }
        }

};


int main(){



    return 0;
}

File: LinkedList/LinkedList/revision/bst.py
import os
from collections import deque

os.system("cls")

class Node:
    def __init__(self, value):
        self.left = None
        self.data = value
        self.right = None
    

class BST:
    def __init__(self):
        self.root = None
        
    def insert(self, value):
        new_node = Node(value)
        
        if(new_node.data > value):
            new_node.left = Node(value)
        else:
            new_node.right = Node(value)
        
    def levelorder_traversal(self):
        q = deque([self.root])
        
        while q:
            levelSize = len(q)
            
            for i in range(levelSize):
                curr = q.popleft()
                
                print(curr.data, end=" ")
                
                if curr.left:
                    q.append(curr.left)
                if curr.right:
                    q.append(curr.right)
            print()
            
bst = BST()

val  = int(input())

while val != -1:
    bst.insert(val)
    val = int(input())
    
bst.levelorder_traversal()        
            